cmd/snapctl/metric.go

```diff
@@ -155,7 +150,42 @@ func getMetric(ctx *cli.Context) error {
 		printFields(w, true, 6, rule.Name, rule.Type, rule.Default, rule.Required, rule.Minimum, rule.Maximum)
 	}
 	w.Flush()
+	return nil
+}
 
+func getMetric(ctx *cli.Context) error {
+	if !ctx.IsSet("metric-namespace") {
+		fmt.Println("namespace is required")
+		fmt.Println("")
+		cli.ShowCommandHelp(ctx, ctx.Command.Name)
+		return errCritical
+	}
+	ns := ctx.String("metric-namespace")
+	ver := ctx.Int("metric-version")
+	metric := pClient.GetMetric(ns, ver)
+	switch mtype := metric.(type) {
+	case []*client.GetMetricResult:
+		// Multiple metrics
+		var merr error
+		for i, m := range metric.([]*client.GetMetricResult) {
+			err := printMetric(m, i)
+			if err != nil {
```

Comment:
I understand your point however why can't we just go through all potential errors and then return instead ?
The only thing this will change is the fact that all errors will be printed at once instead of only one.